testing/matrix-sdk-integration-testing/src/tests/sliding_sync/room.rs

```diff
@@ -1092,6 +1096,90 @@ async fn test_room_preview() -> Result<()> {
     Ok(())
 }
 
+#[async_test]
+async fn test_room_preview_with_room_directory_search_and_room_alias_only() {
+    let server = MatrixMockServer::new().await;
+    let client = server.client_builder().build().await;
+
+    let room_alias = room_alias_id!("#a-room:matrix.org");
+    let expected_room_id = room_id!("!a-room:matrix.org");
+
+    // Allow retrieving the room id
+    server
+        .mock_room_directory_resolve_alias()
+        .ok(expected_room_id.as_ref(), Vec::new())
+        .mock_once()
+        .mount()
+        .await;
+
+    // Given a successful public room search
+    let chunks = vec![PublicRoomsChunkInit {
+        num_joined_members: uint!(0),
+        room_id: expected_room_id.to_owned(),
+        world_readable: true,
+        guest_can_join: true,
+    }
+    .into()];
+    server.mock_public_rooms().ok(chunks, None, None, Some(1)).mock_once().mount().await;
+
+    // The room preview is found
+    let preview = client
+        .get_room_preview(room_alias.into(), Vec::new())
+        .await
+        .expect("room preview couldn't be retrieved");
+    assert_eq!(preview.room_id, expected_room_id);
+}
+
+#[async_test]
+async fn test_room_preview_with_room_directory_search_and_room_alias_only_in_several_homeservers() {
+    let server = MatrixMockServer::new().await;
+    let client = server.client_builder().build().await;
+
+    let room_alias = room_alias_id!("#a-room:matrix.org");
+    let expected_room_id = room_id!("!a-room:matrix.org");
+
+    // Allow retrieving the room id
+    server
+        .mock_room_directory_resolve_alias()
+        .ok(expected_room_id.as_ref(), Vec::new())
+        .mock_once()
+        .mount()
+        .await;
+
+    let via_1 = owned_server_name!("server1.com");
+    let via_2 = owned_server_name!("server2.com");
+
+    // Given a couple of successful public room search responses
+    let via_map = BTreeMap::from_iter(vec![
+        (
+            via_1.to_owned(),
+            vec![PublicRoomsChunkInit {
+                num_joined_members: uint!(0),
+                room_id: expected_room_id.to_owned(),
+                world_readable: true,
+                guest_can_join: true,
+            }
+            .into()],
+        ),
+        (via_2.to_owned(), Vec::new()),
```

Comment:
Can you provide a dummy value for another room, to make sure it's not selected over the correct room?